cmd/boulder-janitor/keyhashtoserial.go

```diff
@@ -0,0 +1,34 @@
+package main
+
+import (
+	"github.com/jmhodges/clock"
+	"github.com/letsencrypt/boulder/db"
+	blog "github.com/letsencrypt/boulder/log"
+)
+
+// newKeyHashToSerialJob returns a batchedDBJob configured to delete expired
+// rows from the keyHashToSerial table.
+func newKeyHashToSerialJob(
+	dbMap db.DatabaseMap,
+	log blog.Logger,
+	clk clock.Clock,
+	config Config) *batchedDBJob {
+	purgeBefore := config.Janitor.KeyHashToSerial.GracePeriod.Duration
+	workQuery := `SELECT id, certNotAfter AS expires FROM keyHashToSerial
+		 WHERE
+		   id > :startID
+		 LIMIT :limit`
```

Comment:
I am not entirely sure how janitor functions, but where does the `limit` get set?

Reply:
The limit comes from the `batchSize` in the config file, e.g.:
https://github.com/letsencrypt/boulder/blob/ff01fe4625a3537c1b44eba9ab6f05f8ff0d77c6/test/config/janitor.json#L12

It gets formatted into the query here:
https://github.com/letsencrypt/boulder/blob/ff01fe4625a3537c1b44eba9ab6f05f8ff0d77c6/cmd/boulder-janitor/job.go#L126-L133